client/packages/glsp-sprotty/src/features/request-response/support.ts

```diff
@@ -0,0 +1,93 @@
+/********************************************************************************
+ * Copyright (c) 2019 EclipseSource and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { inject, injectable } from "inversify";
+import { Action, ActionHandlerRegistry, IActionDispatcher, TYPES } from "sprotty/lib";
+import { v4 as uuid } from 'uuid';
+import { IdentifiableRequestAction, IdentifiableResponseAction, isIdentifiableResponseAction } from "./action-definitions";
+
+@injectable()
+export class RequestResponseSupport {
+    private requestedResponses = new Map<string, Action | undefined>();
+
+    constructor(@inject(TYPES.IActionDispatcher) protected actionDispatcher: IActionDispatcher,
+        @inject(TYPES.ActionHandlerRegistry) protected registry: ActionHandlerRegistry) {
+        registry.register(IdentifiableResponseAction.KIND, this);
+    }
+
+    handle(response: Action) {
+        if (isIdentifiableResponseAction(response)) {
+            const responseId = response.id;
+            if (this.requestedResponses.has(responseId)) {
+                this.requestedResponses.set(responseId, response.action);
+            } else {
+                console.log("[RequestResponse] " + responseId + ": Response without request, ignore.")
```

Comment:
Sprotty provides a lLogger class that can be used to avoid logging directly on the console

Reply:
Thank you for letting me know about this class, I used it in the additional commit! ;-)